crates/host/src/wasmbus/mod.rs

```diff
@@ -3266,7 +3440,36 @@ impl Host {
                         wasmcloud_core::secrets::SecretValue::Bytes(b.to_owned()),
                     ),
                 })
-                .collect(),
+                .collect();
+        // Serializing & sealing an empty map results in a non-empty Vec, which is difficult to tell the
+        // difference between an empty map and an encrypted empty map. To avoid this, we explicitly handle
+        // the case where the map is empty.
+        let source_secrets = if source_secrets_map.is_empty() {
```

Comment:
As much as I don't like `Option<Vec>`s, I think they are probably a better way to indicate the presence or not of secrets. Relying on the length of a Vec means there is a way to shoot ourselves in the foot later. An Option does not.